src/Extend/Mail.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Extend;
+
+use Flarum\Extension\Extension;
+use Illuminate\Contracts\Container\Container;
+
+class Mail implements ExtenderInterface
+{
+    protected $addDrivers = [];
+
+    /**
+     * Get the result of an API request to show a user.
+     *
+     * @param string $identifier Identifier for mail driver. E.g. 'smtp' for SmtpDriver
+     * @param string $driver ::class attribute of driver class, which must implement Flarum\Mail\DriverInterface
+     */
+    public function add(string $identifier, $driver)
```

Comment:
I am thinking this should either be `MailDriver::add` or `Mail::(add)Driver`. `Mail::add` feels a bit too generic.

What do you and others think?

Reply:
Agreed. I'd say either ::driver() or ::addDriver()

Reply:
I went with addDriver in case we want to add more mail extension functionality later. I'd prefer to keep the extenders themselves flexible